app/scripts/modules/core/src/search/infrastructure/SearchResultPodItem.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { MouseEvent } from 'react';
+import { UISref } from '@uirouter/react';
+import { UIRouterContext } from '@uirouter/react-hybrid';
+
+import { IRecentHistoryEntry } from 'core/history';
+import { Tooltip } from 'core/presentation';
+
+import { SearchResult } from './SearchResult';
+
+export interface ISearchResultPodItemProps {
+  categoryName: string;
+  result: IRecentHistoryEntry
+  onRemoveClicked: (categoryName: string, itemId: string) => void;
+}
+
+@UIRouterContext
+export class SearchResultPodItem extends React.Component<ISearchResultPodItemProps> {
+  private handleRemoveItem(evt: MouseEvent<any>, categoryName: string, itemId: string) {
+      evt.preventDefault();
+      this.props.onRemoveClicked(categoryName, itemId);
+  }
+
+  public render() {
+    const { categoryName, result, onRemoveClicked } = this.props;
+    const showRemoveButton = !!onRemoveClicked;
+
+    return (
+      <UISref to={result.state} params={result.params}>
+        <a target="_self" className="list-group-item">
+          <SearchResult item={result}/>
+
+          {showRemoveButton && (
+            <span className="small clickable remove-result-link" onClick={evt => this.handleRemoveItem(evt, categoryName, result.id)} >
```

Comment:
I thought this was bad - does it not create a new function every time the render method is called?